src/models/utils.py

```diff
@@ -33,17 +33,24 @@ def chunk_array(x: Union[Tuple[Union[torch.Tensor, np.ndarray], ...],
         x = (x, )
     x = cast(Tuple[Union[torch.Tensor, np.ndarray]], x)
     num_sections = max(1, x[0].shape[0] // batch_size)
+
     if type(x[0]) == np.ndarray:
         return _chunk_ndarray(x, y, num_sections, shuffle)
     else:
         return _chunk_tensor(x, y, num_sections, shuffle)
 
 
-def _chunk_ndarray(x: Tuple[np.ndarray, ...], y: np.ndarray,
+def _chunk_ndarray(x: Tuple[Optional[np.ndarray], ...], y: np.ndarray,
                    num_sections: int,
-                   shuffle: bool) -> Iterable[Tuple[Tuple[np.ndarray, ...], np.ndarray]]:
+                   shuffle: bool) -> Iterable[Tuple[Tuple[Optional[np.ndarray], ...],
+                                                    np.ndarray]]:
 
-    split_x = [np.array_split(x_section, num_sections) for x_section in x if x_section is not None]
+    split_x = []
+    for idx, x_section in enumerate(x):
+        if x_section is not None:
+            split_x.append(np.array_split(x_section, num_sections))
+        else:
+            split_x.append([None] * num_sections)
```

Comment:
basically we copy the [None] `num_sections` times in order to maintain the structure of the `return_arrays`. I like it